src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2473,21 +2495,23 @@ public bool HasBoxingConversion(TypeSymbol source, TypeSymbol destination, ref H
             if ((source.TypeKind == TypeKind.TypeParameter) &&
                 HasImplicitBoxingTypeParameterConversion((TypeParameterSymbol)source, destination, ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.TypeParameterBoxing;
             }
 
             // The rest of the boxing conversions only operate when going from a value type to a
             // reference type.
             if (!source.IsValueType || !destination.IsReferenceType)
             {
-                return false;
+                return ConversionKind.NoConversion;
             }
 
             // A boxing conversion exists from a nullable type to a reference type if and only if a
             // boxing conversion exists from the underlying type.
             if (source.IsNullableType())
             {
-                return HasBoxingConversion(source.GetNullableUnderlyingType(), destination, ref useSiteDiagnostics);
+                return HasBoxingConversion(source.GetNullableUnderlyingType(), destination, ref useSiteDiagnostics)
```

Comment:
source might be a "T?".  But in that case, even if T is boxable to some destination type, i think it makes sense to classify this as a value-type-boxing since we're boxing ```Nullable<T>``` not T itself.  

What do you think @gafter @AlekseyTs ?